rskj-core/src/test/java/co/rsk/peg/BridgeSupportTest.java

```diff
@@ -4327,6 +4739,130 @@ private Federation getFederation(BridgeConstants bridgeConstants, List<BtcECKey>
+            BtcECKey.ECDSASignature sig2 = new BtcECKey.ECDSASignature(components[0], components[1]).toCanonicalised();
+            bridgeSupport.addSignature(findPublicKeySignedBy(federation.getBtcPublicKeys(), privateKeysToSignWith.get(0)), Lists.newArrayList(sig2.encodeToDER()), keccak256.getBytes());
+        }
+        if (privateKeysToSignWith.size() > 1) {
+            BtcECKey.ECDSASignature sig2 = privateKeysToSignWith.get(1).sign(sighash);
+            byte[] derEncodedSig2 = sig2.encodeToDER();
+            List derEncodedSigs2 = new ArrayList();
+            for (int i = 0; i < numberOfInputsToSign; i++) {
+                derEncodedSigs2.add(derEncodedSig2);
+            }
+            bridgeSupport.addSignature(findPublicKeySignedBy(federation.getBtcPublicKeys(), privateKeysToSignWith.get(1)), derEncodedSigs2, keccak256.getBytes());
+        }
+        bridgeSupport.save();
+        track.commit();
+
+        provider = new BridgeStorageProvider(repository, PrecompiledContracts.BRIDGE_ADDR, bridgeConstants, activationsBeforeForks);
+
+        if ("FullySigned".equals(expectedResult)) {
+            Assert.assertTrue(provider.getRskTxsWaitingForSignatures().isEmpty());
+            Assert.assertThat(logs, is(not(empty())));
+            Assert.assertThat(logs, hasSize(3));
+            LogInfo releaseTxEvent = logs.get(2);
+            Assert.assertThat(releaseTxEvent.getTopics(), hasSize(1));
+            Assert.assertThat(releaseTxEvent.getTopics(), hasItem(Bridge.RELEASE_BTC_TOPIC));
+            BtcTransaction releaseTx = new BtcTransaction(btcParams, ((RLPList) RLP.decode2(releaseTxEvent.getData()).get(0)).get(1).getRLPData());
+            Script retrievedScriptSig = releaseTx.getInput(0).getScriptSig();
+            Assert.assertEquals(4, retrievedScriptSig.getChunks().size());
+            Assert.assertEquals(true, retrievedScriptSig.getChunks().get(1).data.length > 0);
+            Assert.assertEquals(true, retrievedScriptSig.getChunks().get(2).data.length > 0);
+        } else {
+            Script retrievedScriptSig = provider.getRskTxsWaitingForSignatures().get(keccak256).getInput(0).getScriptSig();
+            Assert.assertEquals(4, retrievedScriptSig.getChunks().size());
+            boolean expectSignatureToBePersisted = false; // for "InvalidParameters"
+            if ("PartiallySigned".equals(expectedResult)) {
+                expectSignatureToBePersisted = true;
+            }
+            Assert.assertEquals(expectSignatureToBePersisted, retrievedScriptSig.getChunks().get(1).data.length > 0);
+            Assert.assertEquals(false, retrievedScriptSig.getChunks().get(2).data.length > 0);
+        }
+    }
+
+    private BtcECKey findPublicKeySignedBy(List<BtcECKey> pubs, BtcECKey pk) {
+        for (BtcECKey pub : pubs) {
+            if (Arrays.equals(pk.getPubKey(), pub.getPubKey())) {
+                return pub;
+            }
+        }
+        return new BtcECKey();
```

Comment:
```suggestion
        return pk;
```
We don't want it to create a new Key, we need it to return the same it gets.

Note: This method only makes sense because there are a few tests that compare actual instances (for the logs and such).